### PR TITLE
Add sns endpoint and platform-application sources

### PR DESCRIPTION
**Pull request by tomasdembelli**
Related to #392

### CODE REVIEW

File: sources/sns/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	err = attributes.Set("endpointArn", *input.EndpointArn)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTags, err := tagsByResourceARN(ctx, client, *input.EndpointArn)
```
Comment (dylanratcliffe) [inline sources/sns/endpoint.go:41]: Let's not fail if we can't get tags. Instead if there is an error let's just not set them

File: sources/sns/platform-application.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		Tags:            tagsToMap(resourceTags),
+	}
+
+	// +overmind:link sns-endpoint
+	item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+		Query: &sdp.Query{
+			Type:   "sns-endpoint",
+			Method: sdp.QueryMethod_SEARCH,
+			Query:  *input.PlatformApplicationArn,
+		},
```
Comment (dylanratcliffe) [inline sources/sns/platform-application.go:60]:
```suggestion
			Scope:  scope,
		},
```

File: sources/sns/shared.go [hunk 3]
```diff
@@ -24,7 +24,11 @@ func tagsByResourceARN(ctx context.Context, cli tagLister, resourceARN string) (
 		return nil, err
 	}
 
-	return output.Tags, nil
+	if output != nil && output.Tags == nil {
```
Comment (dylanratcliffe) [inline sources/sns/shared.go:27]: This logic is backwards right? shouldn't it be:

```suggestion
	if output != nil && output.Tags != nil {
```

**Review by dylanratcliffe — CHANGES_REQUESTED** [review]
Couple of small requests, otherwise looking good

File: sources/sns/endpoint.go (hunk 1 above)
Comment (tomasdembelli) [reply sources/sns/endpoint.go]: https://github.com/overmindtech/aws-source/pull/421/commits/51b21c11fb0fee20c9a1318a922f9487b7cd0a7b

File: sources/sns/platform-application.go (hunk 2 above)
Comment (tomasdembelli) [reply sources/sns/platform-application.go:58]: Thanks!

File: sources/sns/shared.go (hunk 3 above)
Comment (tomasdembelli) [reply sources/sns/shared.go:27]: Thanks!

Comment (tomasdembelli) [reply sources/sns/shared.go]: https://github.com/overmindtech/aws-source/pull/421/commits/c4972c6a562315206e62594c0fa0a3ec126c212f

**Review by dylanratcliffe — APPROVED** [review]
Perfect, I'll use a separate branch to get the tests to run and get this merged

**Pull request merged**